### PR TITLE
[new connector] Network Drive

### PR DESCRIPTION
**Pull request by praveen-elastic**
This PR includes:

- Ping functionality for Network Drive, via SMB connection.
- Test cases for Network Drive ping, includes positive and negative test scenarios.
- Added Indexing functionality with test cases.
- Added E2E test cases with docker image.

Note: 
Updated e2e tests to load the data with large files.
We tried to run the e2e test with larger files but we are facing "circuit_breaking_exception"
image

## Connector Checklist:

- [x] Add a module or a directory in [connectors/sources](https://github.com/elastic/connectors-python/tree/main/connectors/sources)
- [x] Implement a class that implements all methods described in connectors.source.BaseDataSource
- [x] Add a unit test in [connectors/sources/tests](https://github.com/elastic/connectors-python/tree/main/connectors/sources/tests) with +90% coverage
- [x] Declare your connector in [config.yml](https://github.com/elastic/connectors-python/blob/main/config.yml) in the sources section
- [x] Declare your dependencies in [requirements.txt](https://github.com/elastic/connectors-python/blob/main/requirements.txt). Make sure you pin these dependencies
- [x] Make sure you use an async lib for your source. If not possible, make sure you don't block the loop
- [x] When possible, provide a docker image that runs the backend service, so we can test the connector. If you can't provide a docker image, provide the credentials needed to run against an online service.
- [x] The test backend needs to return more than 10k documents due to 10k being a default size limit for Elasticsearch pagination. Having more than 10k documents returned from the test backend will help testing connector more deeply

### CODE REVIEW

**Comment by praveen-elastic** [comment]
@tarekziade  Since the original repository has been made public, the fork from which I raised this PR is now detached from origin. Should I close this PR and reopen a new one with these changes ?

File: connectors/sources/network_drive.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        """
+        if (
+            not doit
+            or os.path.splitext(file["title"])[-1] not in SUPPORTED_FILETYPE
+            or not file["size"]
+        ):
+            return
+
+        loop = asyncio.get_running_loop()
+        content = await loop.run_in_executor(
```
[the remark this answers is not in the document]
Comment (tarekziade) [reply connectors/sources/network_drive.py:167]: ok we can stay like this and iterate later for better performances. 2 hours for 78k seems like a lot. Is that something I can test with `make ftest`?

[the remark this answers is not in the document]
Comment (praveen-elastic) [reply connectors/sources/network_drive.py:167]: Sure, i'll try to cover that in ftest.

**Comment by tarekziade** [comment]
I'll go ahead and merge and we can iterate

**Pull request merged**